src/aws-cpp-sdk-core/include/aws/core/http/HttpTypes.h

```diff
@@ -20,43 +20,37 @@ namespace Aws
         /**
          * Models Http methods.
          */
-        enum class HttpMethod
-        {
-            HTTP_GET,
-            HTTP_POST,
-            HTTP_DELETE,
-            HTTP_PUT,
-            HTTP_HEAD,
-            HTTP_PATCH
-        };
-
-        /**
-         * Possible default http factory vended http client implementations.
-         */
-        enum class TransferLibType
-        {
-            DEFAULT_CLIENT = 0,
-            CURL_CLIENT,
-            WIN_INET_CLIENT,
-            WIN_HTTP_CLIENT
-        };
-
-        /**
-         * Configuration for a HTTP client, currently used only by libCurl
-         */
-        enum class TransferLibPerformanceMode
-        {
-            LOW_LATENCY = 0, // run http client for a lower latency, at the expense of CPU
-            REGULAR
-        };
-
-        namespace HttpMethodMapper
-        {
-            /**
-             * Gets the string value of an httpMethod.
-             */
-            AWS_CORE_API const char* GetNameForHttpMethod(HttpMethod httpMethod);
-        } // namespace HttpMethodMapper
+    enum class HttpMethod {
```

Comment:
This looks good. Food for thought:  Maybe in future, we can introduce a wrapped enum macro which can not only generate the enum class but also have a toString method for the concerned enum class. Then we can get the enum string and do contextual parsing as needed.